### PR TITLE
Added Integration Tests Docs

### PR DESCRIPTION
**Pull request by DavidLambauer**

### CODE REVIEW

**Review by johnprendergast — DISMISSED** [review]

File: integration-testing.md [hunk 1]
```diff
@@ -0,0 +1,606 @@
...
+
+The integration test framework creates the temporary test files beneath the
+directory `dev/tests/integration/tmp/sandbox-*` (followed by a long hash ID). To force the test framework to regenerate
+the cache and the other files, remove the directory:
+
+```bash
+rm -r dev/tests/integration/tmp/sandbox-*
+```
+
+<a href="#rabbitmq-management-port-constant"></a>
```
Comment (michielgerritsen) [inline integration-testing.md:194]: I don't think that mentioning the `RABBITMQ_MANAGEMENT_PORT` constant here adds value to the document.

File: integration-testing.md [hunk 2]
```diff
@@ -0,0 +1,606 @@
...
+#### RABBITMQ_MANAGEMENT_PORT Constant
+
+The default value for the `RABBITMQ_MANAGEMENT_PORT` constant is `15672`. The RabbitMQ Management Plugin APIs are
+accessible on port 15672 by default. This constant provides the configuration value when the test environment is not
+using the default. Tests depending on `Magento\TestFramework\Helper\Amqp`, often used for validating queue and exchange
+status, depend on the management plugin being enabled. Beginning in 2.3.1, the helper no longer assumes RabbitMQ is
+running on localhost and will use the connection configuration defined in `env.php`.
+
+<a href="#php-memory-limit"></a>
```
Comment (michielgerritsen) [inline integration-testing.md:203]: I think that managing the memory_limit is out of scope for this subject.

File: integration-testing.md [hunk 3]
```diff
@@ -0,0 +1,606 @@
...
+
+Remember to replace the example database, username, and password with ones that suit your requirements and conventions.
+
+<a href="#configuring-the-framework-for-test-environment"></a>
+### Configuring the Framework for Test Environment
+
+The Magento 2 integration test framework provides a configuration file template located
+at `mage2ce/dev/tests/integration/etc/install-config-mysql.php.dist`.
+
+Copy this file to `mage2ce/dev/tests/integration/etc/install-config-mysql.php` (without the `.dist` suffix) and add your
```
Comment (michielgerritsen) [inline integration-testing.md:129]: The `mage2ce` has not added value, all other paths in this document are without a prefix.

File: integration-testing.md [hunk 4]
```diff
@@ -0,0 +1,606 @@
...
+Running Unit Tests in PhpStorm for instructions on creating a basic run configuration. Then, configure the integration
+test to use the configuration file.
+
+<a href="#using-the-integration-test-configuration-file"></a>
+### Using the Integration Test Configuration File
+
+The only difference in the run configuration is that the integration test `phpunit.xml.dist` or `phpunit.xml`
+configuration file from the `dev/tests/integration` directory must be selected.
+
+<a href="#integration-tests-file-structure"></a>
```
Comment (michielgerritsen) [inline integration-testing.md:370]: I think that this section needs to be either revisited or removed:
- The `Magento` folder does not exist.
- `bootstrap.php` does not exist either.
- It is missing the `_files` folder.

My vote would be to remove this.

File: integration-testing.md [hunk 5]
```diff
@@ -0,0 +1,606 @@
...
+- `Magento/` – A set of classes that implement the Magento integration tests framework.
+- `bootstrap.php` – The PHPUnit bootstrap script.
+- `etc/install-config-<db_vendor>.php` – A configuration file that provides values for installing the Magento
+  application.
+- `testsuite/` – The test suite.
+- `tmp/` – A writable directory for storing temporary data during test execution.
+- `sandbox-<hash>/` – The folder where each Magento instance stores temporary and configuration data.
+- `phpunit.xml.dist` – A PHPUnit configuration file.
+
+<a href="#using-data-providers-in-integration-tests"></a>
```
Comment (michielgerritsen) [inline integration-testing.md:386]: Although it's a nice feature of PHPUnit, data providers don't belong in this document. I think it leads the reader away from the subject of running tests.

File: integration-testing.md [hunk 6]
```diff
@@ -0,0 +1,606 @@
...
+
+In Magento 2, fixtures are PHP scripts that set required preconditions. Here is an example of a fixture script:
+
+```php
+<?php
+require 'default_rollback.php';
+require __DIR__ . '/../../Customer/_files/customer_rollback.php';
+require __DIR__ . '/../../Store/_files/second_website_with_two_stores_rollback.php';
+require __DIR__ . '/../../Store/_files/second_website_with_store_group_and_store_rollback.php';
+```
```
Comment (michielgerritsen) [inline integration-testing.md:461]: The section above shows a rollback. I think it's better to update the example to maybe something like this:

```php
<?php
$objectManager = \Magento\TestFramework\Helper\Bootstrap::getObjectManager();
/** @var $repository \Magento\Customer\Api\CustomerRepositoryInterface */
$repository = $objectManager->create(\Magento\Customer\Api\CustomerRepositoryInterface::class);
$customer = $objectManager->create(\Magento\Customer\Model\Customer::class);

// Set customer attributes like name, email, specific group, etc.

$repository->save($customer);
```

File: integration-testing.md [hunk 7]
```diff
@@ -0,0 +1,606 @@
...
+## Understanding DocBlock Annotations in Magento 2 Integration Tests
+
+In Magento 2 integration tests, DocBlock annotations are used to provide metadata about the test, such as preconditions,
+expectations, and dependencies. Here are some of the most commonly used annotations:
+
+<a href="#magentodatafixture"></a>
+### @magentoDataFixture
+
+This annotation is used to specify a fixture script that sets up the necessary preconditions for the test. The script is
+run before the test is executed.
```
Comment (michielgerritsen) [inline integration-testing.md:512]: > The base path for this annotation is always `dev/tests/integration/testsuite`.

File: integration-testing.md [hunk 8]
```diff
@@ -0,0 +1,606 @@
...
+
+```php
+/**
+ * @magentoDataFixture Magento/Catalog/_files/categories.php
+ */
+public function testMethod()
+{
+    // Test code here
+}
+```
```
Comment (michielgerritsen) [inline integration-testing.md:522]: > The `@magentoDataFixture Magento/Catalog/_files/categories.php` annotation will resolve to `dev/tests/integration/testsuite/Magento/Catalog/_files/categories.php`.

File: integration-testing.md [hunk 9]
```diff
@@ -0,0 +1,606 @@
...
+In the context of testing, a fixture is a fixed state of a set of objects used as a baseline for running tests. The
+purpose of a test fixture is to ensure that there is a well-known and fixed environment in which tests are run so that
+results are repeatable. Examples of fixtures:
+
+- Preparation of input data and setup/creation of fake or mock objects
+- Loading a database with a specific, known set of data
+- Copying a specific known set of files creating a test fixture will create a set of objects initialized to certain
+  states.
+
+<a href="#creating-fixtures"></a>
```
Comment (michielgerritsen) [inline integration-testing.md:450]: I think if there is a section on how to create a fixture, there should also be a path where to place it. But I have no clue where that should be. I guess in the `dev/tests/integration/testsuite` folder somewhere? I have never done this personally.

**Review by michielgerritsen — COMMENTED** [review]
I have added a bunch of suggestions. Feel free to ignore them if you don't agree :-)

File: integration-testing.md (hunk 1 above)
Comment (DavidLambauer) [reply integration-testing.md:194]: Kinda agree.

File: integration-testing.md (hunk 3 above)
Comment (DavidLambauer) [reply integration-testing.md:129]: Yep!

File: integration-testing.md (hunk 6 above)
Comment (DavidLambauer) [reply integration-testing.md]: There are many default fixtures you can use. It is often faster to use a default one to create any object I want to test against. I updated this part to have it more understandable.

File: integration-testing.md (hunk 9 above)
Comment (DavidLambauer) [reply integration-testing.md]: It couldn't be uglier. You have to add your whole relative path to it. 

```php
/**
 * @magentoDataFixture ../../../../app/code/Vendor/Module/Test/Integration/_files/my_custom_fixture.php
 */
public function testMethod()
{
    // Test code here
}

```

**Comment by DavidLambauer** [comment]
@michielgerritsen feel free to do a quick review again. I updated the docs according to your feedback 🥳

**Pull request merged**